shell/src/main/java/alluxio/cli/fs/FileSystemShell.java

```diff
@@ -84,6 +89,13 @@ protected String getShellName() {
 
   @Override
   protected Map<String, Command> loadCommands() {
-    return FileSystemShellUtils.loadCommands(FileSystemContext.create(mConfiguration));
+    mContext = FileSystemContext.create(mConfiguration);
+    return FileSystemShellUtils.loadCommands(mContext);
+  }
+
+  @Override
+  public void close() throws IOException {
+    super.close();
```

Comment:
an simpler way can be introduce a new member `Closer` in `AbstractShell`, and add resource (e.g., the `mContext` here to the closer from `AbstractShell`. Then you can directly close this `Closer` in `AbstractShell.close()` rather than having the same code on close repeatedly across different impl of `AbstractShell`

Reply:
Thanks Bin. Closer made it much simpler and cleaner.